test/extended/templates/templateinstance_cross_namespace.go

```diff
@@ -124,19 +131,53 @@ var _ = g.Describe("[sig-devex][Feature:Templates] templateinstance cross-namesp
 
 		// wait for garbage collector to do its thing
 		err = wait.Poll(100*time.Millisecond, 30*time.Second, func() (bool, error) {
-			_, err = cli.TemplateClient().TemplateV1().TemplateInstances(cli.Namespace()).Get(context.Background(), templateinstance.Name, metav1.GetOptions{})
+			t, err := cli.TemplateClient().TemplateV1().TemplateInstances(cli.Namespace()).Get(context.Background(), templateinstance.Name, metav1.GetOptions{})
 			if kerrors.IsNotFound(err) {
 				return true, nil
 			}
+
+			tiJSON, _ := json.MarshalIndent(t, "", "    ")
+			framework.Logf("Template Instance object during deletion : %s", string(tiJSON))
+			s1, e1 := cli.KubeClient().CoreV1().Secrets(cli.Namespace()).Get(context.Background(), "secret1", metav1.GetOptions{})
+			if e1 != nil {
+				framework.Logf("error secret1 during deletion: %s", e1.Error())
+			} else {
+				s1JSON, _ := json.MarshalIndent(s1, "", "    ")
+				framework.Logf("secret1 during deletion: %s", string(s1JSON))
+			}
+			s2, e2 := cli.KubeClient().CoreV1().Secrets(cli2.Namespace()).Get(context.Background(), "secret2", metav1.GetOptions{})
+			if e2 != nil {
+				framework.Logf("error secret2 during deletion: %s", e2.Error())
+			} else {
+				s2JSON, _ := json.MarshalIndent(s2, "", "    ")
+				framework.Logf("secret2: during deletion: %s", string(s2JSON))
+			}
+
 			return false, err
 		})
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		_, err = cli.KubeClient().CoreV1().Secrets(cli.Namespace()).Get(context.Background(), "secret1", metav1.GetOptions{})
-		o.Expect(kerrors.IsNotFound(err)).To(o.BeTrue())
+		// poll in case the dependent objects deletions in the template instance controller finalizer complete with
+		// unexpected timings compared to the template instance itself; as you'll see in the debub prints above,
+		// there is not use of `ownerReferences` with templates ... we delete dependent objects, much like we delete
+		// operands today from our CVO operator finalizers
```

Comment:
nit - this comment could be more succinct.
```suggestion
		// Template instances use a finalizer to delete objects, rather than owner references. As a result, objects
		// created by the template can be deleted after the TemplateInstance itself is deleted. 
```